### PR TITLE
Court class detail views

### PR DESCRIPTION
**Pull request by actlikewill**
- This adds court class detail views along with some refactoring of the court views.

https://www.loom.com/share/cd7f7fc1d5e84a2a84260674a2458160?sid=0ab87cb8-ed23-45ae-a6e3-66ced1967695

closes laws-africa/kenyalaw-pj#31

### CODE REVIEW

File: peachjam/templates/peachjam/_court_list.html [hunk 1]
```diff
@@ -12,6 +12,10 @@ <h4 class="mb-2">{{ court_class }}</h4>
               </li>
             {% endfor %}
           </ul>
+          {% if court_class.show_listing_page %}
+            <a class="btn btn-sm btn-info"
+               href="{% url 'court_class' court_class.slug %}">View All</a>
```
Comment (longhotsummer) [inline peachjam/templates/peachjam/_court_list.html:17]: Translatable, and use sentence case: "View all"

File: peachjam/views/courts.py [hunk 2]
```diff
@@ -150,78 +90,150 @@ def grouped_judgments(self, documents):
         return [{"key": key, "judgments": list(group)} for key, group in groups]
 
 
-class CourtYearView(CourtDetailView):
+class CourtDetailView(BaseJudgmentFilterView):
+    template_name = "peachjam/court_detail.html"
+
+    @property
```
Comment (longhotsummer) [inline peachjam/views/courts.py:96]: Use `@cached_property` so we go to the db only once

File: peachjam/views/courts.py [hunk 3]
```diff
@@ -1,69 +1,36 @@
...
+from peachjam.models import Court, CourtClass, CourtRegistry, Judgment
 from peachjam.views.generic_views import FilteredDocumentListView
 
 
-class CourtDetailView(FilteredDocumentListView):
-    """List View class for filtering a court's judgments.
-
-    This view may be restricted to a specific year.
-    """
+class BaseJudgmentFilterView(FilteredDocumentListView):
```
Comment (longhotsummer) [inline peachjam/views/courts.py:12]: minor: maybe `FilteredJudgmentListView` is a more consistent name.

File: peachjam/views/courts.py [hunk 4]
```diff
@@ -1,69 +1,36 @@
...
-        if "year" in self.kwargs:
-            qs = qs.filter(date__year=self.kwargs["year"])
-
-        if "month" in self.kwargs:
-            qs = qs.filter(date__month=self.kwargs["month"])
+    queryset = Judgment.objects.prefetch_related("judges", "labels")
 
-        return qs
+    def base_view_name(self):
+        return "Judgments"
```
Comment (longhotsummer) [inline peachjam/views/courts.py:20]: I'm guessing this is used in a title or something, so it must be translateable.

**Comment by longhotsummer** [comment]
Nice!

File: peachjam/views/courts.py (hunk 2 above)
Comment (actlikewill) [reply peachjam/views/courts.py]: added

File: peachjam/views/courts.py (hunk 3 above)
Comment (actlikewill) [reply peachjam/views/courts.py]: changed

File: peachjam/templates/peachjam/_court_list.html [hunk 5]
```diff
@@ -12,6 +12,10 @@ <h4 class="mb-2">{{ court_class }}</h4>
               </li>
             {% endfor %}
           </ul>
+          {% if court_class.show_listing_page %}
+            <a class="btn btn-sm btn-primary"
```
Comment (longhotsummer) [inline peachjam/templates/peachjam/_court_list.html:16]: I recommend we make this a simple link below the h4 heading, with `class="mb-3"` and text "All judgments"

Comment (actlikewill) [reply peachjam/templates/peachjam/_court_list.html]: good idea. this is how it looks now
image

**Review by longhotsummer — APPROVED** [review]
Nice!

**Pull request merged**